index.js

```diff
@@ -13,60 +25,117 @@ const QUESTIONS = [{
     }]
 }];
 
+// Use the body-parser middleware to parse the request body
+app.use(bodyParser.json())
 
-const SUBMISSION = [
-
-]
+const SUBMISSIONS = [];
 
 app.post('/signup', function(req, res) {
+  const USERS = []
   // Add logic to decode body
-  // body should have email and password
-
+  const { email, password, isAdmin } = req.body
 
-  //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
+  // Check if the user with the given email already exists
+  const userExists = USERS.some(user => user.email === email)
+  if (userExists) {
+    // If the user already exists, return 409 Conflict status code
+    res.status(409).send('User already exists!')
+    return
+  }
 
+  // Store email and password in the USERS array
+  USERS.push({ email, password, isAdmin })
 
-  // return back 200 status code to the client
-  res.send('Hello World!')
-})
+  // Return back 200 status code to the client
+  res.sendStatus(200)
+});
 
 app.post('/login', function(req, res) {
   // Add logic to decode body
-  // body should have email and password
+  const { email, password } = req.body
 
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
+  // Find the user with the given email in the USERS array
+  const user = USERS.find(user => user.email === email)
 
+  // Check if a user was found
+  if (!user) {
+    // If the user doesn't exist or the password is incorrect, return 401 Unauthorized status code
+    res.status(401).send('Invalid email!')
+    return
+  }
 
-  // If the password is the same, return back 200 status code to the client
-  // Also send back a token (any random string will do for now)
-  // If the password is not the same, return back 401 status code to the client
+  // Check if the password is correct
+  if (user.password !== password) {
+    // If the user doesn't exist or the password is incorrect, return 401 Unauthorized status code
+    res.status(401).send('Wrong password!')
+    return
+  }
 
+  // If the user exists and the password is correct, generate a random token string for authentication
+  const token = Math.random().toString(36).substring(7)
 
-  res.send('Hello World from route 2!')
-})
+  // Store the token in the user object for future authentication
+  user.token = token
 
-app.get('/questions', function(req, res) {
+  // Return back 200 status code to the client along with the token
+  res.status(200).json({ token, isAdmin: user.isAdmin })
+});
 
+
+app.get('/questions', function(req, res) {
   //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
+  res.json(QUESTIONS);
+})
+
+// leaving as hard todos
+
+// Create a route that lets an admin add a new problem
+// ensure that only admins can do that.
+
+app.post("/questions", function(req, res) {
+  // Add logic to decode body
+  const { title, description, testCases } = req.body
```

Comment:
you should input token in the body and verify it and find the corresponding user then see if its an admin